core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -53,81 +57,115 @@
     public AmplifyConfiguration(@NonNull Map<String, CategoryConfiguration> configs) {
+        this(configs, new LinkedHashMap<>());
+    }
+
+    private AmplifyConfiguration(
+            Map<String, CategoryConfiguration> configs,
+            Map<Platform, String> platformVersions
+    ) {
         this.categoryConfigurations = new HashMap<>();
         this.categoryConfigurations.putAll(configs);
+        this.platformVersions = platformVersions;
     }
 
     /**
      * Build an {@link AmplifyConfiguration} directly from an {@link JSONObject}.
-     * Users should prefer loading from resources files via {@link #fromConfigFile(Context)},
-     * or {@link #fromConfigFile(Context, int)}.
+     *
      * @param json A JSON object
      * @return An AmplifyConfiguration
      * @throws AmplifyException If the JSON does not represent a valid AmplifyConfiguration
+     * @deprecated This method is now deprecated in favor of {@link #builder(JSONObject)}.
      */
+    @Deprecated
     @NonNull
     public static AmplifyConfiguration fromJson(@NonNull JSONObject json) throws AmplifyException {
-        final List<CategoryConfiguration> possibleConfigs = Arrays.asList(
-            new AnalyticsCategoryConfiguration(),
-            new ApiCategoryConfiguration(),
-            new AuthCategoryConfiguration(),
-            new DataStoreCategoryConfiguration(),
-            new HubCategoryConfiguration(),
-            new LoggingCategoryConfiguration(),
-            new PredictionsCategoryConfiguration(),
-            new StorageCategoryConfiguration()
-        );
-
-        final Map<String, CategoryConfiguration> actualConfigs = new HashMap<>();
-        try {
-            for (CategoryConfiguration possibleConfig : possibleConfigs) {
-                String categoryJsonKey = possibleConfig.getCategoryType().getConfigurationKey();
-                if (json.has(categoryJsonKey)) {
-                    possibleConfig.populateFromJSON(json.getJSONObject(categoryJsonKey));
-                    actualConfigs.put(categoryJsonKey, possibleConfig);
-                }
-            }
-        } catch (JSONException error) {
-            throw new AmplifyException(
-                "Could not parse amplifyconfiguration.json ",
-                error, "Check any modifications made to the file."
-            );
-        }
-        return new AmplifyConfiguration(Immutable.of(actualConfigs));
+        return builder(json).build();
     }
 
     /**
      * Creates an AmplifyConfiguration from an amplifyconfiguration.json file.
+     *
      * @param context Context needed for reading JSON file
      * @return An Amplify configuration instance
      * @throws AmplifyException If there is a problem in the config file
+     * @deprecated This method is now deprecated in favor of
+     *          {@link AmplifyConfiguration#builder(Context)}.
      */
     @SuppressWarnings("WeakerAccess")
+    @Deprecated
```

Comment:
I think it's still useful; I'd leave it, and use it around the code base.

Reply:
Replacing it with `@VisibleForTesting` for now because it really shouldn't be used by the customers.